MH-12692: update maven bundle plugin for java8

pmiddend opened the pull request:
This commit updates the maven-bundle-plugin to 3.x, which updates the
“bnd” utility used to resolve osgi dependencies.

By upgrading the plugin, it fixes a bug (a known bug triggering an
`ArrayOutOfBoundsException`) which prevented Java 8 from being used in
production code. However, it also introduced an ambiguity. Both
“jsr311” and “jsr339” provide the Java package
“javax.rs.ws” (apparently, JSR-331 is JAX-RS 1.1 and JSR-339 is
version 2.0). This lead to an ambiguous import chain, for example in
`matterhorn-admin-ui-ng`.

The solution employed here is to force OSGI to use version 2 of the
“javax.rs.ws” package (and replacing a dependency in the
feature.xml). All the code using JAX-RS still compiles and runs.

This also fixes a bug in the pom of wowza, where using
`${opencast.version}` lead to a `NumberFormatException`.

Also, I couldn’t resist my OCD and fixed one or two indentation
mishaps in the vicinity of the bundle-plugin directives I was
changing.

This work is sponsored by plapadoo.

staubesv commented:
@pmiddend To me, it looks very much as JSR-339 would replace JSR-331 (both JAX-RS). Is there a particular reason to keep JSR-331 at all?

pmiddend commented:
Yeah, that's what I thought, too. I see two problems with this, though: first, I don't know which package is adequate for providing JSR-339. There's at least the one from servicekit and one called `javax.ws.rs-api`. Secondly, some dependencies of Opencast will pull in jsr311 (maven central lists >1.000 usages), so it won't purge the ambiguity, at least not until those packages use an updated JSR.

lkiesow commented:
I took a short look at it and it looks good to me. Though, this does not seem to be something we really need (should?) deploy into a maintenance release, is it? Can we redirect this to `develop`?

@staubesv, while this might not be perfect, it definitely seems to be an improvement. I would merge this once it's redirected against `develop` if you approve.

staubesv commented:
@lkiesow Absolutely fine for me to merge this into develop... just wondered JSR-339 couldn't replaced JSR-311, but @pmiddend provided a reasonable explanation. Thanks Plapadoo!

Commit:
MH-12692: Update maven bundle plugin for java8

This commit updates the maven-bundle-plugin to 3.x, which updates the
“bnd” utility used to resolve osgi dependencies.

By upgrading the plugin, it fixes a bug (a known bug triggering an
ArrayOutOfBoundsException) which prevented Java 8 from being used in
production code. However, it also introduced an ambiguity. Both
“jsr311” and “jsr339” provide the Java package
“javax.rs.ws” (apparently, JSR-331 is JAX-RS 1.1 and JSR-339 is
version 2.0). This lead to an ambiguous import chain, for example in
matterhorn-admin-ui-ng.

The solution employed here is to force OSGI to use version 2 of the
“javax.rs.ws” package (and replacing a dependency in the
feature.xml). All the code using JAX-RS still compiles and runs.

This also fixes a bug in the pom of one of the bundles, where using
${opencast.version} lead to a NumberFormatException.

Also, I couldn’t resist my OCD and fixed one or two indentation
mishaps in the vicinity of the bundle-plugin directives I was
changing.

pmiddend commented:
I understand your concerns. After some manual fiddling with what `git diff` outputs, I was able to redirect this to `develop`.

Edit: Sorry, I screwed up, fixing...

lkiesow commented:
Merged into `develop`, this works as expected. @staubesv, if you have no objection I'm going to merge this tomorrow

staubesv commented:
No objections from my side.

lkiesow commented:
Merged with commit 5d159a8df

KatrinIhler commented:
Looks like this update to Java8 might be breaking the creation of series, since that currently doesn't work in develop (try at develop.opencast.org), but it does work for me if I revert this commit, see the issue MH-12713.
